docs/README.md

```diff
@@ -0,0 +1,624 @@
+
+TSG reserves the right to determine what constitutes a breaking changes. A definition of "breaking changes" will be made available before formal release.
+
+# HTTP API
+
+## API Discovery
+
+In order to reduce client coupling, the AIM API provides an API discovery document available at:
+
+<a class="discovery-config-url"></a>
+
+<pre><code id="discovery-config"></code></pre>
+
+The primary attributes of interest is the `urls` object, which provides static names to full or partial URLs.
+
+## Authentication
+
+The AIM API leverages [Firebase Authentication](https://firebase.google.com/docs/auth) to securely authenticate services and users. To enable API usage, service accounts are created and used to generate a secret Refresh Token that is given to the primary API contact.
+
+These Refresh Tokens do not expire and can be used to retrieve short lived ID Tokens. ID Tokens are used to directly communicate with the AIM API, which will validate the ID Token.
+
+In short, the authentication flow looks like the following:
+1. Exchange a Refresh Token for an ID Token with Firebase
+2. Use the ID Token with the AIM API
+3. Repeat step 1 as the previous ID Token expires
+
+![Authentication Flow](./authentication_flow.png)
+
+Once acquired, the ID Token must be sent in the `Authorization` HTTP Header as a `Bearer` token.
+
+As ID Tokens are short lived, a new one must be fetched before expiration and replaced in requests to the API. ID Tokens are [JSON Web Tokens](https://jwt.io/), so any standard JWT libary can be used to decode them and inspect the `exp` entry for a Unix timestamp after which the token will be rejected by the API. A number of JWT libraries are referenced in the link above.
+
+### Obtain an ID Token
+
+In order to obtain an ID Token, use the `id_token` url from the [Discovery document](#api-discovery), which allows us to exchange our Refresh Token for a fresh ID Token.
+
+<a class="id_token-url"></a>
+
+Make a POST request with the following payload, injecting the Refresh Token as specified: `{"grant_type": "refresh_token", "refresh_token": <API Key>}`. Extract the `id_token` field from the response, which contains the ID Token, which can then be sent to the API.
+
+For example, with `curl` to make the request and `jq` to extract the field:
+
+```sh
+ID_TOKEN_URL="<ID Token Discovery URL>"
+API_KEY="<Refresh Token>"
+curl -fsSl -XPOST \
+     -H "Content-Type: application/json" \
+     -d "{\"grant_type\": \"refresh_token\", \"refresh_token\": \"$API_KEY\"}" \
+     "$ID_TOKEN_URL" \
+     | jq -r '.id_token'
+```
+
+You now have an ID Token that can be used with the AIM API!
+
+### Abuse and Privacy
+
+In order to prevent abuse and data leaks, Refresh Tokens must be stored securely. In particular, avoid unneeded sharing of Refresh Tokens or storing them in source files/source control.
```

Comment:
Al:
> Why state “avoid unneeded” and state in caps DON’T SHARE?

Reply:
I mainly said this because I knew we'd be sharing them over email to start and figured there'd be sharing to get it to the right folks in the company. In the future, this should be done from an (unbuilt) "Make an API key" page on the website.

Can you give me a full sentence to use? How do you want to resolve the tense change in the second half (since we're going present progressive -> present in the first half).